src/networking.c

```diff
@@ -2779,9 +2785,10 @@ static void setProtocolError(const char *errstr, client *c) {
             p++;
         }
 
+    log_client_and_proto_info:
         /* Log all the client and protocol info. */
-        int loglevel = (c->flag.primary) ? LL_WARNING : LL_VERBOSE;
-        serverLog(loglevel, "Protocol error (%s) from client: %s. %s", errstr, client, buf);
+        serverLog(loglevel, "Protocol error (%s) from client: %s. %s", errstr, client,
+                  server.hide_user_data_from_log ? "*redacted*" : buf);
```

Comment:
When hide-user-data is disabled, the logged messages look like these, depending on buffer length:

* `"Protocol error (errstr) from client: (client). Query buffer during protocol error: (buf)"`

When hide-user-data is enabled, this code logs

* `"Protocol error (errstr) from client: (client). *redacted*"`
 
The message doesn't say what is redacted. It doesn't say that it's the query buffer.

Another thing that bothers me a little is that "protocol error" appears twice in the logged message. I mean in the old messages. This wasn't well formulated.

To align the two variants and also get rid of the duplicated wording, I think we should remove "Query buffer during protocol error" from the `snprintf` code above (where `buf` is constructed) and add "query buffer" here instead, i.e.

```suggestion
        serverLog(loglevel, "Protocol error (%s) from client: %s. Query buffer: %s", errstr, client,
                  server.hide_user_data_from_log ? "*redacted*" : buf);
```